### PR TITLE
Add support for GitHub Actions

### PR DESCRIPTION
**Pull request by TobiGr**
#### What is it?
- [ ] Bug fix (user facing)
- [ ] Feature (user facing)
- [ ] Code base improvement (dev facing)
- [x] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
This enables GitHub Actions to work with PRs.
Build PRs pointing to dev or master branch and create a comment linking to the APK download page on success. Otherwise, the creator is notified (if enabled by them).
Example PR: https://github.com/TobiGr/NewPipe/pull/7

#### ToDo
- [ ] Check out the Git branch and do not use HEAD. Currently, all APKs have the id `org.schabi.newpipe.HEAD` and app name is "NewPipe HEAD"
- [ ] Disable & Remove Travis integration
- [ ] Require successful GitHub Action checks for merging a PR

#### Ideas
- Do not comment a link to the workflow page when `<!-- NO_APK_NOTIFICATION -->` is in (preferably at the end of) the PR description.
- Post a warning when a PR is approved, but the extractor repo is not set to `TeamNewPipe/NewPipeExtractor`
#### Fixes the following issue(s)

- Addresses #3940 

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

**Comment by wb9688** [comment]
Something to check: if I change the workflow file in my PR, will it use the workflow file from my PR or from the branch it will merge into? If the first, it could be possible to change the workflow file to print `secrets.GITHUB_TOKEN`, which you obviously don't want.

**Comment by ingingin** [comment]
The grammar on that bot isn't perfect

**Comment by yausername** [comment]
I opened a PR on @TobiGr 's fork, it runs the modified workflow file but fails to post comment.
The `secrets.GITHUB_TOKEN` given to forked PRs has read-only access to the origin project. Hence posting comment fails.

File: .github/workflows/build-pull-requests.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Upload APK
+        uses: actions/upload-artifact@v2
+        with:
+          name: app
+          path: app/build/outputs/apk/debug/*.apk
+
+      - name: Create comment linking to the artifact
+        uses: thollander/actions-comment-pull-request@1f25fabed60c3f141743c3a522529950a8fb2191
+        with:
+          message: 'The APK was build successfuly. You can find it here: https://github.com/TobiGr/NewPipe/actions/runs/${{github.run_id}}'
```
Comment (TobiGr) [inline .github/workflows/build-pull-requests.yml:33]: This line needs to be changed

File: .github/workflows/build-pull-requests.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+on:
+  pull_request:
+    branches: # we might want to add all branches here
+      - 'master'
+      - 'dev'
+
+jobs:
+  build:
+    name: Build project & generate APK
+    runs-on: ubuntu-18.04
```
Comment (mhmdanas) [inline .github/workflows/build-pull-requests.yml:12]: Any reason not to use `ubuntu` instead?

**Comment by B0pol** [comment]
https://www.zdnet.com/article/google-to-github-times-up-this-unfixed-high-severity-security-bug-affects-developers

**Comment by opusforlife2** [comment]
> https://www.zdnet.com/article/google-to-github-times-up-this-unfixed-high-severity-security-bug-affects-developers

I suppose we should wait, then.

**Comment by FireMasterK** [comment]
Having an action without the depreciated `add-path` or `set-env` commands should be safe according to https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/.

**Comment by XiangRongLin** [comment]
TravisCI changed their pricing last month, giving public repos only 1000min/month https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing#building-on-a-public-repositories-only.

I doubt it is enough with each build taking 6-7min, meaning ~150builds. Meaning the migration to GitHub actions would rise in importance

**Comment by TobiGr** [comment]
@XiangRongLin Thanks for letting us know!

I just rebased my commit. I'll try to find a solution for the mentioned problems.

File: .github/workflows/build-pull-requests.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: set up JDK 1.8
+        uses: actions/setup-java@v1.4.0
```
Comment (FireMasterK) [inline .github/workflows/build-pull-requests.yml:17]: Change this to v1.4.3.
v.1.4.0 uses the vulnerable `set-env` command.

**Comment by FireMasterK** [comment]
I've created a pull request to just replace Travis: #5132

**Comment by TobiGr** [comment]
Closing in favour of #5132. See my comments regarding debug APKs in that thread.

**Pull request closed without merging**